.github/workflows/ci_cmake.yml

```diff
@@ -0,0 +1,135 @@
+# CI workflow for wxWidgets builds using CMake.
+name: CMake builds
+
+on:
+  push:
+    branches:
+      - master
+    paths-ignore:
+      - '.github/workflows/ci_msw_cross.yml'
+      - 'build/tools/appveyor*.bat'
+      - 'build/tools/travis-ci.sh'
+      - 'distrib/**'
+      - 'docs/**'
+      - 'interface/**'
+      - 'include/msvc/**'
+      - 'include/wx/msw/**'
+      - 'locale/**'
+      - 'src/msw/**'
+      - '*.md'
+      - '*.yml'
+      - 'wxwidgets.props'
+  pull_request:
+    branches:
+      - master
+    paths-ignore:
+      - '.github/workflows/ci_msw_cross.yml'
+      - 'build/tools/appveyor*.bat'
+      - 'build/tools/travis-ci.sh'
+      - 'distrib/**'
+      - 'docs/**'
+      - 'interface/**'
+      - 'include/msvc/**'
+      - 'include/wx/msw/**'
+      - 'locale/**'
+      - 'src/msw/**'
+      - '*.md'
+      - '*.yml'
+      - 'wxwidgets.props'
+
+jobs:
+  build:
+    runs-on: ${{ matrix.runner }}
+    name: ${{ matrix.name }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: Ubuntu 18.04 wxGTK 3
+            runner: ubuntu-18.04
+            cmake_generator: Unix Makefiles
+          - name: macOS 10.15 wxOSX
+            runner: macos-10.15
+            cmake_generator: Xcode
+            cmake_defines: -DCMAKE_CXX_STANDARD=11
+          - name: macOS 10.15 wxIOS
+            runner: macos-10.15
+            cmake_generator: Xcode
+            cmake_defines: -DCMAKE_SYSTEM_NAME=iOS -DCMAKE_FIND_ROOT_PATH=/usr/local -DCMAKE_XCODE_ATTRIBUTE_CODE_SIGNING_ALLOWED=NO
+            cmake_samples: OFF
+            cmake_tests: OFF
+
+    env:
+      wxGTK_VERSION: ${{ matrix.gtk_version && matrix.gtk_version || 3 }}
+      wxCMAKE_GENERATOR: ${{ matrix.cmake_generator }}
+      wxCMAKE_DEFINES: ${{ matrix.cmake_defines }}
+      wxCMAKE_TESTS: ${{ matrix.cmake_tests }}
+      wxCMAKE_SAMPLES: ${{ matrix.cmake_samples }}
```

Comment:
As for those two, I think it's somewhat confusing to define them here when their actual values are set below, so I think it could be better to do something like
```sh
cmake_tests=${{ matrix.cmake_tests }}
if [ -z $cmake_tests ]; then cmake_tests=CONSOLE_ONLY; fi
```
below.